browser/net/brave_network_delegate_base.cc

```diff
@@ -46,18 +46,13 @@ BraveNetworkDelegateBase::BraveNetworkDelegateBase(
   // Initialize the preference change registrar.
   base::PostTaskWithTraits(
       FROM_HERE, {BrowserThread::UI},
-      base::Bind(&BraveNetworkDelegateBase::InitPrefChangeRegistrar,
-                 base::Unretained(this)));
-  // Retrieve the current referral headers, if any.
-  base::PostTaskWithTraits(
-      FROM_HERE, {BrowserThread::UI},
-      base::Bind(&BraveNetworkDelegateBase::GetReferralHeaders,
+      base::Bind(&BraveNetworkDelegateBase::InitPrefChangeRegistrarOnUI,
```

Comment:
I think the problem here is that this really doesn't seem like the right place for this stuff in the first place. I think it should be in `BraveReferralsService` and it can provide a static method that can be accessed from the IO thread. Since this is local_state it isn't profile-specific so we can track it globally

Reply:
Yeah, this is exactly what I mentioned in the TODO comment. I propose to merge this until I refactor the whole thing globally.

Reply:
sorry, missed that. I'm good with merging and follow-up refactor as well